lib/resty/upstream/healthcheck.lua

```diff
@@ -214,64 +211,53 @@ local function check_peer(ctx, id, peer, is_backup)
         return
     end
 
+    -- define shortcut error function
+    local report_error = function(...) 
```

Comment:
I do understand your intention here but we have other concerns, apparently. Another approach is to explicitly pass the `sock`, `peer`, `is_backup` and etc values as arguments of this `report_error` function (which should reside in the top-level scope). It may not be ideal but it can considered a "improvement" :)